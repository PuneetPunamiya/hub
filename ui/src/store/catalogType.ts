import { types } from "mobx-state-tree";
export const CatalogTypeList = [
  { name: "official", selected: false },
  { name: "verified", selected: false },
  { name: "community", selected: false }
];

const Type = types
  .model({
    name: types.string,
    selected: false
  })
  .actions(self => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export const CatalogTypeStore = types
  .model({
    catalogtypelist: types.array(Type)
  })
  .views(self => ({
    get count() {
      return self.catalogtypelist.length;
    },
    get catalogType() {
      const { catalogtypelist } = self;
      return catalogtypelist
        .filter((type: any) => type.selected === true)
        .map((type: any) => type.name);
    }
  }))
  .actions(self => ({
    add(item: any) {
      self.catalogtypelist.push(item);
    },
    setSelectedCatalogType(kindName: any) {
      self.catalogtypelist.forEach((kind: any) => {
        if (kind.name === kindName) {
          kind.toggle();
          return;
        }
      });
    }
  }))
  .actions(self => ({
    loadCatalogType() {
      CatalogTypeList.forEach((kind: any) => self.add(kind));
    }
  }))
  .actions(self => ({
    afterCreate() {
      self.loadCatalogType();
    }
  }));
