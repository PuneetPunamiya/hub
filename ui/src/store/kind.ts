import { types } from "mobx-state-tree";

const Kind = types
	.model({
		name: types.string,
		selected: false,
	})
	.actions((self) => ({
		toggle() {
			self.selected = !self.selected;
		},
	}));

export const KindStore = types
	.model({
		kind: types.array(Kind),
	})
	.views((self) => ({
		get count() {
			return self.kind.length;
		},
	}))
	.actions((self) => ({
		add(item: any) {
			self.kind.push(item);
		},
	}));
