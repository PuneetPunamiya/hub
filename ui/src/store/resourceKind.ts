import { types } from "mobx-state-tree";

export const KindList = [
  { name: "task", selected: false },
  { name: "pipeline", selected: false }
];

const Kind = types
  .model({
    name: types.string,
    selected: false
  })
  .actions(self => ({
    toggle() {
      self.selected = !self.selected;
    }
  }));

export const ResourceKindStore = types
  .model({
    resourcekindlist: types.array(Kind)
  })
  .views(self => ({
    get count() {
      return self.resourcekindlist.length;
    },
    get resourceKind() {
      const { resourcekindlist } = self;
      return resourcekindlist
        .filter((type: any) => type.selected)
        .map((type: any) => type.name);
    }
  }))
  .actions(self => ({
    add(item: any) {
      self.resourcekindlist.push(item);
    },
    setSelectedKind(kindName: any) {
      self.resourcekindlist.forEach((kind: any) => {
        if (kind.name === kindName) {
          kind.toggle();
          return;
        }
      });
    }
  }))
  .actions(self => ({
    loadKind() {
      KindList.forEach((kind: any) => self.add(kind));
    }
  }))
  .actions(self => ({
    afterCreate() {
      self.loadKind();
    }
  }));
