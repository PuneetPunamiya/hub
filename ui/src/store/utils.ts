export function assert(value: unknown): asserts value {
  if (value === undefined) {
    throw new Error('value must be defined');
  }
}

export enum DropDownItems {
  Name = 'Name',
  Rating = 'Rating'
}
