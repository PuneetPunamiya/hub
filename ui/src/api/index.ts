import axios from "axios";
import { API_URL } from "../config/constants";
import { ICategory } from "../store/category";
import { IResource } from "../store/resources";
export interface Api {
  categories(): Promise<ICategory>;
  resources(): Promise<IResource>;
}
export class Hub implements Api {
  async resources() {
    try {
      return axios.get(
        `https://api-tekton-hub-staging.apps.openshift-web.p0s5.p1.openshiftapps.com/resources`
      );
    } catch (err) {
      return err.response;
    }
  }
  async categories() {
    try {
      return axios.get(`${API_URL}/categories`);
    } catch (err) {
      return err.response;
    }
  }
}
