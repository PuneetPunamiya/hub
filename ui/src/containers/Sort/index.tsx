import React, { useState } from 'react';
import { DropdownItem, Dropdown, DropdownToggle } from '@patternfly/react-core';
import { useObserver } from 'mobx-react';
import './Sort.css';
import { useMst } from '../../store/root';

const Sort: React.FC = () => {
  const { resources } = useMst();

  const items: Array<string> = Object.values(resources.dropDownItems);

  const [sort, setSort] = useState('Name');
  const [isOpen, set] = useState(false);

  const dropDownItems = items.map((value) => (
    <DropdownItem key={value} onClick={(e) => setSort(e.currentTarget.id)}>
      {value}
    </DropdownItem>
  ));

  const onToggle = (isOpen: React.SetStateAction<boolean>) => set(isOpen);
  const onSelect = () => set(!isOpen);

  return useObserver(() => {
    return (
      <div>
        <Dropdown
          onSelect={onSelect}
          toggle={<DropdownToggle onToggle={onToggle}>{sort}</DropdownToggle>}
          isOpen={isOpen}
          dropdownItems={dropDownItems}
          className="hub-sort-dropdown"
        />
      </div>
    );
  });
};

export default Sort;
