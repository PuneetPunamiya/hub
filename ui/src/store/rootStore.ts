import { types } from "mobx-state-tree";
import fuzzysort from "fuzzysort";
import { ResourceStore } from "./resources";
import { ResourceKindStore } from "./resourceKind";
import { CatalogTypeStore } from "./catalogType";
import { CategoryStore } from "./category";
import { Resource } from "./resources";
import { SearchStore } from "./search";

export const RootStore = types
  .model("RootStore", {
    resourcestore: types.optional(ResourceStore, {}),
    resourcekindstore: types.optional(ResourceKindStore, {}),
    catalogtypestore: types.optional(CatalogTypeStore, {}),
    categorystore: types.optional(CategoryStore, {}),
    dashboardresource: types.optional(types.array(Resource), []),
    searchstore: types.optional(SearchStore, {})
  })
  .views(self => ({
    get alldashboardresource() {
      const { resources } = self.resourcestore;
      const { filteredTags } = self.categorystore;
      const { catalogType } = self.catalogtypestore;
      const { resourceKind } = self.resourcekindstore;
      const { searchtext } = self.searchstore;

      return filterResourceByAllFilter(
        resources,
        filteredTags,
        catalogType,
        resourceKind,
        searchtext
      );
    }
  }))

  .actions(self => ({
    afterCreate() {
      self.resourcestore.loadResources();
      self.categorystore.loadCategories();
    }
  }));

const filterResourceByAllFilter = (
  resources: any,
  filteredTags: any,
  catalogType: any,
  resourceKind: any,
  searchtext: any
) => {
  let filterresult = resources;
  if (resourceKind.length > 0) {
    let tempfilterresult: any = [];
    resourceKind.forEach((kind: any) => {
      filterresult.forEach((resource: any) => {
        if (resource.type === kind) {
          tempfilterresult.push(resource);
        }
      });
    });
    filterresult = tempfilterresult;
  }
  if (catalogType.length > 0) {
    let tempfilterresult: any = [];
    catalogType.forEach((catalogtype: any) => {
      filterresult.forEach((resource: any) => {
        if (resource.type === catalogtype) {
          tempfilterresult.push(resource);
        }
      });
    });
    filterresult = tempfilterresult;
  }
  if (filteredTags.length > 0) {
    let tempfilterresult: any = [];
    filteredTags.forEach((tag: any) => {
      filterresult.forEach((r: any) => {
        r.tags.forEach((t: any) => {
          if (t.name === tag) {
            tempfilterresult.push(r);
          }
        });
      });
    });
    filterresult = tempfilterresult;
  }
  if (searchtext !== "") {
    const tempfilterresult = fuzzysort.go(searchtext, resources, {
      keys: ["name", "displayName"]
    });
    filterresult = tempfilterresult;
  }
  return filterresult;
};
