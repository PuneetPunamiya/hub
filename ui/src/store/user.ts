import { flow, getEnv, Instance, types } from 'mobx-state-tree';
import { Api } from '../api';

export const TokenInfo = types.model({
  token: types.optional(types.string, ''),
  expiresAt: types.optional(types.number, 0),
  refreshInterval: types.optional(types.string, '')
});

export type IAuthStore = Instance<typeof AuthStore>;
export type ITokenInfo = Instance<typeof TokenInfo>;
export interface AuthCodeProps {
  code: string;
}

export const AuthStore = types
  .model({
    accessTokenInfo: types.optional(TokenInfo, {}),
    refreshTokenInfo: types.optional(TokenInfo, {}),
    isLoading: false,
    isAuthenticated: false,
    err: ''
  })
  .actions((self) => ({
    addAccessTokenInfo(item: ITokenInfo) {
      self.accessTokenInfo.token = item.token;
      self.accessTokenInfo.expiresAt = item.expiresAt;
      self.accessTokenInfo.refreshInterval = item.refreshInterval;
    },
    addRefreshTokenInfo(item: ITokenInfo) {
      self.refreshTokenInfo.token = item.token;
      self.refreshTokenInfo.expiresAt = item.expiresAt;
      self.refreshTokenInfo.refreshInterval = item.refreshInterval;
    },
    setIsAuthenticated(l: boolean) {
      self.isAuthenticated = l;
    },
    setIsLoading(l: boolean) {
      self.isLoading = l;
    },
    onFailure(err: string) {
      self.err = err;
    }
  }))
  .views((self) => ({
    get api(): Api {
      return getEnv(self).api;
    }
  }))
  .actions((self) => ({
    authenticate: flow(function* (authCode: AuthCodeProps) {
      try {
        const { api } = self;

        const json = yield api.authentication(authCode.code);

        self.addAccessTokenInfo(json.data.access);
        self.addRefreshTokenInfo(json.data.refresh);

        self.setIsAuthenticated(true);
        self.setIsLoading(true);
      } catch (err) {
        self.err = err.toString();
      }
    })
  }));
