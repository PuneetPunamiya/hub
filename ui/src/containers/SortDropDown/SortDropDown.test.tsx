import React from 'react';
import { mount } from 'enzyme';
import Sort from '.';
import { Dropdown } from '@patternfly/react-core';
import { FakeHub } from '../../api/testutil';
import { createProviderAndStore } from '../../store/root';
import { assert } from '../../store/utils';

const TESTDATA_DIR = `src/store/testdata`;
const api = new FakeHub(TESTDATA_DIR);
const { Provider } = createProviderAndStore(api);

describe('Sort DropDown', () => {
  it('should render the dropdown', () => {
    const component = mount(
      <Provider>
        <Sort />
      </Provider>
    );

    const item = component.find(Dropdown).props().dropdownItems;
    assert(item);

    expect(item.length).toBe(2);
    expect(item[0].key).toBe('Name');
    expect(component.debug()).toMatchSnapshot();
  });
});
