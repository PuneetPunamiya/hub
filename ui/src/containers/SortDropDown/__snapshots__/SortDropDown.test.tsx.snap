// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort DropDown should render the dropdown 1`] = `
"<Provider>
   
  <Sort>
    <div>
      <Dropdown onSelect={[Function: onSelect]} toggle={{...}} isOpen={false} dropdownItems={{...}} className=\\"hub-sort-dropdown\\">
        <DropdownWithContext toggle={{...}} isOpen={false} dropdownItems={{...}} className=\\"hub-sort-dropdown\\" isPlain={false} isGrouped={false} position=\\"left\\" direction=\\"down\\" onSelect={[Function: onSelect]} autoFocus={true} menuAppendTo=\\"inline\\">
          <div className=\\"pf-c-dropdown hub-sort-dropdown\\" data-ouia-component-type=\\"PF4/Dropdown\\" data-ouia-safe={true} data-ouia-component-id={0}>
            <DropdownToggle onToggle={[Function: onToggle]} parentRef={{...}} getMenuRef={[Function (anonymous)]} isOpen={false} id=\\"pf-dropdown-toggle-id-0\\" isPlain={false} aria-haspopup={true} onEnter={[Function: onEnter]}>
              <Toggle onEnter={[Function: onEnter]} id=\\"pf-dropdown-toggle-id-0\\" className=\\"\\" isOpen={false} parentRef={{...}} getMenuRef={[Function (anonymous)]} isActive={false} isDisabled={false} isPlain={false} isPrimary={false} onToggle={[Function: onToggle]} aria-haspopup={true} isSplitButton={false} bubbleEvent={false}>
                <button id=\\"pf-dropdown-toggle-id-0\\" className=\\"pf-c-dropdown__toggle\\" type=\\"button\\" onClick={[Function: onClick]} aria-expanded={false} aria-haspopup={true} onKeyDown={[Function: onKeyDown]} disabled={false}>
                  <span className=\\"pf-c-dropdown__toggle-text\\">
                    Name
                  </span>
                  <span className=\\"pf-c-dropdown__toggle-icon\\">
                    <CaretDownIcon color=\\"currentColor\\" size=\\"sm\\" noVerticalAlign={false}>
                      <svg style={{...}} fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 320 512\\" aria-labelledby={{...}} aria-hidden={true} role=\\"img\\">
                        <path d=\\"M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z\\" transform=\\"\\" />
                      </svg>
                    </CaretDownIcon>
                  </span>
                </button>
              </Toggle>
            </DropdownToggle>
          </div>
        </DropdownWithContext>
      </Dropdown>
    </div>
  </Sort>
   
</Provider>"
`;
